src/infql.jl

```diff
@@ -508,49 +503,120 @@ function getproperty(F::QL{T, AdaptiveQLFactors{T}, AdaptiveQLTau{T}}, d::Symbol
     end
 end
 
-@inline function getQ(F::QL{T, AdaptiveQLFactors{T}, AdaptiveQLTau{T}}) where T
-    Q = AdaptiveQLQ{T}(zeros(1,1),F.factors,F.τ,(0,0))
-    # initialize as 50x50, then adaptively expand
-    resizedata!(Q,50)
-    return Q
-end
+# TODO: adaptively build L*b using caching and forward-substitution
+*(L::LowerTriangular{T, AdaptiveQLFactors{T}}, b::LayoutVector) where T = ApplyArray(*, L, b)
 
-mutable struct AdaptiveQLQ{T} <: AbstractCachedMatrix{T}
-    data::Matrix{T}
-    factors::AdaptiveQLFactors{T}
-    τ::AdaptiveQLTau{T}
-    datasize::Tuple{Int, Int}
-    AdaptiveQLQ{T}(D, f, tau, N::Tuple{Int, Int}) where T = new{T}(D, f, tau, N)
-end
+MemoryLayout(::AdaptiveQLFactors) = LazyBandedLayout()
+bandwidths(F::AdaptiveQLFactors) = bandwidths(F.data)
 
-size(::AdaptiveQLQ) = (ℵ₀, ℵ₀)
 
-function resizedata!(K::AdaptiveQLQ, nm...)
-    nm = maximum(nm)
-    νμ = K.datasize[1]
-    if nm > νμ
-        cache_filldata!(K, νμ:nm)
-        K.datasize = size(K.data).÷2
+# Q = \\prod_{i=1}^{\\min(m,n)} (I - \\tau_i v_i v_i^T)
+@inline getQ(F::QL{T, AdaptiveQLFactors{T}, AdaptiveQLTau{T}}) where T = QLPackedQ(F.factors,F.τ)
+
+getindex(Q::QLPackedQ{T,<:AdaptiveQLFactors{T}}, i::Int, j::Int) where T =
+(Q'*[Zeros{T}(i-1); one(T); Zeros{T}(∞)])[j]'
+getindex(Q::QLPackedQ{<:Any,<:AdaptiveQLFactors}, I::AbstractVector{Int}, J::AbstractVector{Int}) =
+    [Q[i,j] for i in I, j in J]
```

Comment:
Ditto

Reply:
ok seems like for the getindex we cannot use the default methods, have to keep these to make it work.